### DISCUSSION TITLE
How to read a JSON or CSV document from memory?

### DISCUSSION THREAD

**Discussion by not-my-profile** [apache/datafusion#6963] [Q&A]
The format-specific `read_` methods form [SessionContext](https://docs.rs/datafusion/27.0.0/datafusion/execution/context/struct.SessionContext.html) only accept paths, so I am wondering how you can instead provide something that implements [std::io::Read](https://doc.rust-lang.org/stable/std/io/trait.Read.html).

Apparently back in 5.0.0 there were some [`try_new_from_reader` methods](https://docs.rs/datafusion/5.0.0/datafusion/index.html?search=try_new_from_reader) but they have been removed and I am struggling to figure out how to do the same with the new API.

**Comment by alamb** [comment]
the idea is that you would implement something that implemented the [`ObjectStore` trait](https://docs.rs/object_store/0.6.1/object_store/trait.ObjectStore.html) and then read a "path" from that 

 Here is an example of one that does so for in memory blobs  https://docs.rs/object_store/0.6.1/object_store/memory/struct.InMemory.html

**Comment by Jefffrey** [comment]
This is indeed a limitation of DataFusion at the moment. I'll close this discussion in favour of the issue opened to track this: https://github.com/apache/datafusion/issues/18802
